File: tools/ghidra_scripts/DecompAll.py
# Ghidra Jython postScript: 反编译所有函数到文件。用法 -postScript DecompAll.py <out.c>
from ghidra.app.decompiler import DecompInterface
from ghidra.util.task import ConsoleTaskMonitor

args = getScriptArgs()
out = args[0] if args else "decomp.c"
fm = currentProgram.getFunctionManager()
di = DecompInterface()
di.openProgram(currentProgram)
mon = ConsoleTaskMonitor()
f = open(out, "w")
n = 0
for fn in fm.getFunctions(True):
    try:
        res = di.decompileFunction(fn, 60, mon)
        if res and res.decompileCompleted():
            f.write("// ===== %s @ %s  size=%d =====\n" % (fn.getName(), fn.getEntryPoint(), fn.getBody().getNumAddresses()))
            f.write(res.getDecompiledFunction().getC())
            f.write("\n\n")
            n += 1
    except Exception as e:
        pass
f.close()
print("DECOMP_DONE functions=%d -> %s" % (n, out))

File: tools/ghidra_scripts/DecompAll.java
import ghidra.app.script.GhidraScript;
import ghidra.app.decompiler.DecompInterface;
import ghidra.app.decompiler.DecompileResults;
import ghidra.program.model.listing.Function;
import ghidra.program.model.listing.FunctionIterator;
import ghidra.util.task.ConsoleTaskMonitor;
import java.io.PrintWriter;
import java.io.FileWriter;

public class DecompAll extends GhidraScript {
    public void run() throws Exception {
        String[] args = getScriptArgs();
        String out = (args.length > 0) ? args[0] : "decomp.c";
        DecompInterface di = new DecompInterface();
        di.openProgram(currentProgram);
        PrintWriter w = new PrintWriter(new FileWriter(out));
        int n = 0;
        ConsoleTaskMonitor mon = new ConsoleTaskMonitor();
        FunctionIterator it = currentProgram.getFunctionManager().getFunctions(true);
        while (it.hasNext()) {
            Function fn = it.next();
            try {
                DecompileResults res = di.decompileFunction(fn, 60, mon);
                if (res != null && res.decompileCompleted()) {
                    w.println("// ===== " + fn.getName() + " @ " + fn.getEntryPoint()
                              + " size=" + fn.getBody().getNumAddresses() + " =====");
                    w.println(res.getDecompiledFunction().getC());
                    w.println();
                    n++;
                }
            } catch (Exception e) { /* skip */ }
        }
        w.close();
        println("DECOMP_DONE functions=" + n + " -> " + out);
    }
}
